src/main/kotlin/ro/hibyte/betting/entity/BetTemplate.kt

```diff
@@ -0,0 +1,72 @@
+package ro.hibyte.betting.entity
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import jakarta.persistence.*
+import ro.hibyte.betting.exceptions.types.BadRequestException
+import ro.hibyte.betting.repository.BetTemplateRepository
+
+
+@Entity
+data class BetTemplate (
+    @Id
+    @GeneratedValue
+    var id: Long? = null,
+
+    var name: String,
+
+    @Enumerated(EnumType.STRING)
+    var type: BetTemplateType,
+
+    @JsonIgnore
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "betTemplate", cascade = [CascadeType.ALL])
+    var betTypes: List<BetType> = ArrayList(),
+
+    @ElementCollection
+    @CollectionTable(name = "bet_template_options", joinColumns = [JoinColumn(name = "bet_template_id")])
+    @Column(name = "option")
+    var multipleChoiceOptions: List<String> = ArrayList()
+
+) {
+
+
+    companion object {
+        //this method ensures that no MULTIPLE_CHOICE BetTemplates can be created if the number of options is less than 2
```

Comment:
I just noticed this: does this PR also include the changes from #11 ? If so, please ignore the comments left on the things that are not part of #3 (if any)